private/pkg/protosource/option_extension_descriptor.go

```diff
@@ -126,3 +126,16 @@ func isDescendantPath(descendant, ancestor []int32) bool {
 	}
 	return true
 }
+
+func (o *optionExtensionDescriptor) PresentFields() map[protoreflect.FieldDescriptor]protoreflect.Value {
+	// Note: This does not bother to handle unrecognized fields.
+	// Should not be a problem since descriptors models in the buf CLI codebase should have them
+	// all correctly parsed and known.
+	msg := o.message.ProtoReflect()
+	fields := map[protoreflect.FieldDescriptor]protoreflect.Value{}
```

Comment:
This might happen to work, but `protoreflect.FieldDescriptor` is an interface, and (1) isn't guaranteed to be comparable (2) is likely just comparing struct pointers. If this method needs to be added, something like `ForEachPresentField(func(protoreflect.FieldDescriptor, protoreflect.Value))` may be more appropriate.

Reply:
I really like this idea, to provide a callback-based range function instead of building and returning a map. 👍 

But also know that, technically, this _is_ specified to work in the package docs. Search for "Each descriptor is comparable." in the docs: https://pkg.go.dev/google.golang.org/protobuf/reflect/protoreflect. While it is possible that you could have two distinct values represent the same semantic field, that's not possible here because the message can only have a single `protoreflect.FieldDescriptor` and value per tag number.

Reply:
Ugh I hate it regardless - I'm on mobile, I can't find it, but I believe you. But still ugh.

If you're ok with the callback approach, I'd prefer that, so that we can keep "no interfaces as map keys" as a good mantra to maintain in general.

Reply:
Yes, I do like the idea. Let's switch to the callback approach.